v0/destinations/snapchat_custom_audience/transform.js

```diff
@@ -0,0 +1,226 @@
+const sha256 = require("sha256");
+const {
+  defaultRequestConfig,
+  removeUndefinedAndNullValues,
+  simpleProcessRouterDest,
+  CustomError
+} = require("../../util");
+const ErrorBuilder = require("../../util/error");
+const { BASE_URL } = require("./config");
+
+/**
+ * Get access token to be bound to the event req headers
+ *
+ * Note:
+ * This method needs to be implemented particular to the destination
+ * As the schema that we'd get in `metadata.secret` can be different
+ * for different destinations
+ *
+ * @param {Object} metadata
+ * @returns
+ */
+const getAccessToken = metadata => {
+  // OAuth for this destination
+  const { secret } = metadata;
+  // we would need to verify if secret is present and also if the access token field is present in secret
+  if (!secret || !secret.access_token) {
+    throw new ErrorBuilder()
+      .setStatus(500)
+      .setMessage("Empty/Invalid access token")
+      .build();
+  }
+  return secret.access_token;
+};
+
+const addUserResponseBuilder = (metadata, message, { Config }) => {
+  const { schema, audienceId } = Config;
+  //   const resp = constructPayload(
+  //     message,
+  //     mappingConfig[ConfigCategory.AUDIENCE_LIST.name]
+  //   );
+
+  const payload = { schema: [], data: [] };
+  let schemaType;
+  let hashedProperty;
+  switch (schema) {
+    case "email":
+      schemaType = "EMAIL_SHA256";
+      hashedProperty = message.properties.listData.add.email;
+      hashedProperty = sha256(hashedProperty.toLowerCase().trim());
+      break;
+    case "phone":
+      schemaType = "PHONE_SHA256";
+      hashedProperty =
+        message.properties.listData.add.phone ||
+        message.properties.listData.add.mobile;
+      hashedProperty = sha256(
+        hashedProperty
+          .toLowerCase()
+          .trim()
+          .replace(/^0+/, "")
+          .replace(/[^0-9]/g, "")
+      );
+      break;
+    case "mobileAdId":
+      schemaType = "MOBILE_AD_ID_SHA256";
+      hashedProperty =
+        message.properties.listData.add.mobileId ||
+        message.properties.listData.add.mobileAdId ||
+        message.properties.listData.add.mobile_id;
+      hashedProperty = sha256(hashedProperty.toLowerCase());
+      break;
+    default:
+      throw new CustomError("Invalid schema", 400);
+  }
+
+  payload.schema.push(schemaType);
+  payload.data.push(hashedProperty);
+
+  const response = defaultRequestConfig();
+
+  response.endpoint = `${BASE_URL}/segments/${audienceId}/users`;
+  response.body.JSON = removeUndefinedAndNullValues(payload);
+  const accessToken = getAccessToken(metadata);
+  //   response.params = { listId: Config.listId, customerId: filteredCustomerId };
+  response.headers = {
+    Authorization: `Bearer ${accessToken}`,
+    "Content-Type": "application/json"
+  };
+
+  return response;
+};
+
+const removeUserResponseBuilder = (metadata, message, { Config }) => {
```

Comment:
The basic payload for both the remove and add are the same. Only the REST API method is different i.e DELETE and POST respectively.

Reply:
Removed